pkg/policy/api/rule_validation.go

```diff
@@ -309,6 +309,14 @@ func (pr *L7Rules) sanitize() error {
 	}
 
 	if pr.DNS != nil {
+		if len(ports) == 0 {
+			return fmt.Errorf("Port 53 must be specified for DNS rules")
+		}
+		for _, port := range ports {
+			if port.Port != "53" {
+				return fmt.Errorf("DNS rules are only allowed on port 53")
```

Comment:
I don't think we have the reason for this written down anywhere. It might be useful to note it here and when it could be changed, but that might be more clear once https://github.com/cilium/cilium/pull/6921 is merged.

Reply:
I'll add a comment and fire another test run.